src/gui/statusbar.cpp

```diff
@@ -251,32 +251,10 @@ void StatusBar::updateAltSpeedsBtn(bool alternative)
     refresh();
 }
 
-void StatusBar::capDownloadSpeed()
+void StatusBar::capSpeed()
 {
-    BitTorrent::Session *const session = BitTorrent::Session::instance();
-
-    bool ok = false;
-    const long newLimit = SpeedLimitDialog::askSpeedLimit(
-                parentWidget(), &ok, tr("Global Download Speed Limit"), session->downloadSpeedLimit());
-    if (ok)
-    {
-        qDebug("Setting global download rate limit to %.1fKb/s", newLimit / 1024.);
-        session->setDownloadSpeedLimit(newLimit);
-        refresh();
-    }
-}
-
-void StatusBar::capUploadSpeed()
-{
-    BitTorrent::Session *const session = BitTorrent::Session::instance();
-
-    bool ok = false;
-    const long newLimit = SpeedLimitDialog::askSpeedLimit(
-                parentWidget(), &ok, tr("Global Upload Speed Limit"), session->uploadSpeedLimit());
-    if (ok)
-    {
-        qDebug("Setting global upload rate limit to %.1fKb/s", newLimit / 1024.);
-        session->setUploadSpeedLimit(newLimit);
-        refresh();
-    }
+    auto dialog {new SpeedLimitDialog(this)};
+    dialog->setAttribute(Qt::WA_DeleteOnClose);
+    dialog->open();
+    refresh();
```

Comment:
I don't think this line has any useful effect now, try removing it.

Reply:
Indeed, removed.